com.unity.ml-agents/Runtime/Grpc/RpcCommunicator.cs

```diff
@@ -504,7 +504,7 @@ public static byte[] GetSideChannelMessage(Dictionary<int, SideChannel> sideChan
                         var messageList = sideChannel.MessageQueue;
                         foreach (var message in messageList)
                         {
-                            binaryWriter.Write(sideChannel.ChannelId);
+                            binaryWriter.Write(sideChannel.ChannelId.ToByteArray());
```

Comment:
was this part of the problem you were running into with GUID from C# --> Python?

Reply:
Yes. I ended-up fixing on the python side